client/lib/media/actions.js

```diff
@@ -37,7 +37,11 @@ MediaActions.setQuery = function( siteId, query ) {
 	} );
 };
 
-MediaActions.fetch = function( siteId, itemId ) {
+// TODO: Once this entire module is converted to Redux,
+// `receiveMediaItems` shouldn't be passed as an argument anymore,
+// but directly imported and dispatch()ed from inside `fetch()`,
+// which needs to be turned into a Redux thunk.
+MediaActions.fetch = function( siteId, itemId, receiveMediaItems = null ) {
```

Comment:
It wasn't very difficult to get me convinced :) 

Okay, let's reopen and rebase #2167 and prepare it for merging. We can then use it for such cases like in this PR. @aduth @ockham what do you folks think? Is there anything that prevents us from doing it at this time?